core/server/server.js

```diff
@@ -137,12 +138,11 @@ const publicConfigSchema = Joi.object({
     teamcity: defaultService,
     trace: Joi.boolean().required(),
   }).required(),
-  cacheHeaders: {
-    defaultCacheLengthSeconds: Joi.number().integer().required(),
-  },
+  cacheHeaders: { defaultCacheLengthSeconds: nonNegativeInteger },
```

Comment:
This change is not strictly related to this PR but I noticed it here. I don't think it ever makes sense to say cache something for a negative number of seconds.